### PR TITLE
Remove the lock.

### PR DESCRIPTION
**Pull request by p-lr**

### CODE REVIEW

File: app/src/main/java/com/qozix/tileview/tiles/TileCanvasViewGroup.java [hunk 1]
```diff
@@ -207,7 +209,7 @@ void renderTiles() {
   }
 
   private boolean establishOpaqueRegion() {
-    boolean shouldInvalidate = false;
+    boolean shouldInvalidate = mTilesInCurrentViewport.isEmpty();
     for( Tile tile : mTilesInCurrentViewport ) {
```
Comment (p-lr) [inline app/src/main/java/com/qozix/tileview/tiles/TileCanvasViewGroup.java:213]: Everything is here. I will explain the details for that change soon.

**Comment by p-lr** [comment]
Fixes #3.

File: app/src/main/java/com/qozix/tileview/TileView.java [hunk 2]
```diff
@@ -827,7 +834,9 @@ public void onZoomUpdate( float scale, Origination origin ) {
 
   @Override
   public void onZoomEnd( float scale, Origination origin ) {
-    mDetailLevelManager.unlockDetailLevel();
+    if ( origin == null ) {
```
Comment (moagrius) [inline app/src/main/java/com/qozix/tileview/TileView.java:837]: on end, do we care about origin?  should we always assume we should resume render...?

Comment (p-lr) [inline app/src/main/java/com/qozix/tileview/TileView.java:837]: We could do that

**Comment by moagrius** [comment]
should we also remove this line https://github.com/moagrius/TileViewLabs/blob/missing-tiles-from-cao/app/src/main/java/com/qozix/tileview/tiles/TileCanvasViewGroup.java#L165

**Comment by p-lr** [comment]
> should we also remove this line https://github.com/moagrius/TileViewLabs/blob/missing-tiles-from-cao/app/src/main/java/com/qozix/tileview/tiles/TileCanvasViewGroup.java#L165

I think i removed it

**Comment by moagrius** [comment]
you did - sorry!

**Comment by moagrius** [comment]
i've added you as a collaborator - i understand and like what you're doing - i was going to suggest we deprecate the whole lock detail level api, but i suppose it's fine to have that as an option if someone wants to use it - so i'm +1 on this, feel free to make any changes you want and merge...  after that, i'm good to start migrating?

**Comment by moagrius** [comment]
(of course, i'll definitely still review, pull, test, so anything else you want to document is great - but i also bet it's very late for you and don't want to hang you up)

**Comment by p-lr** [comment]
> i've added you as a collaborator - i understand and like what you're doing

Thank you

>  i was going to suggest we deprecate the whole lock detail level api, but i suppose it's fine to have that as an option if someone wants to use it - so i'm +1 on this, feel free to make any changes you want and merge... after that, i'm good to start migrating?

I was thinking the same thing, but i agree that it could serve someone.

**Comment by p-lr** [comment]
> after that, i'm good to start migrating?

After that, i've not seen any noticeable issue, for instance at least. So agree for migrating.

**Comment by moagrius** [comment]
awesome, thanks peter - i'll post a PR on the main repo when ready.  let me know if there's anything else

**Comment by p-lr** [comment]
Ok!

**Pull request merged**